seat/src/contract.rs

```diff
@@ -29,11 +29,13 @@ pub fn instantiate(
     let hub_contract = mut_deps.branch().api.addr_validate(&msg.hub_contract)?;
     HUB_CONTRACT.save(mut_deps.storage, &hub_contract)?;
     // instantiate all modules
-    let mut modules = SeatModules::new(mut_deps.branch().as_ref());
+    let mut modules = SeatModules::new();
     let res = modules.instantiate(mut_deps.branch(), env, info.clone(), &msg);
     set_contract_version(mut_deps.storage, CONTRACT_NAME, CONTRACT_VERSION).unwrap();
     CONFIG.save(mut_deps.storage, &Config { owner: info.sender })?;
-    res
+    Ok(res.unwrap().add_event(
```

Comment:
In general, we don't want to unwrap results in cosmwasm contracts, as that causes information-less panics. This should use `?` or a different type of err check